### PR TITLE
Feature lb network

### PR DESCRIPTION
**Pull request by drmaas**
Have the floating IP get pulled from the pool or allocated based on the network id used.

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy [hunk 1]
```diff
@@ -346,6 +347,34 @@ abstract class OpenstackClientProvider {
...
+   * Get an unallocated IP from the network, or if none are found, try to create a new floating IP in the network.
+   * @param region
+   * @param networkName
+   * @return
+   */
+  FloatingIP getOrCreateFloatingIp(final String region, final String networkName) {
+    handleRequest {
+      FloatingIP ip = getRegionClient(region).compute().floatingIps().list().find { !it.fixedIpAddress }
+      if (!ip) {
+        ip = client.useRegion(region).compute().floatingIps().allocateIP(networkName)
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: Will this throw if it fails to allocate, or return `null`?

**Comment by lwander** [comment]
LGTM post comment

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy (hunk 1 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: Good question, it throws a ClientResponseException with a status code of 403, which we catch and rethrow as an OpenstackProviderException.

Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: In that case, shouldn't this throw an exception if `ip` is `null` here?

Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: IP can be null from the first call. The second call shouldn't return null if it allocates a new IP without throwing an exception. I can add a check though just to be sure.

**Pull request merged**